### PR TITLE
PWX-27934: fix CSI socket path is not mounted in portworx container

### PR DESCRIPTION
**Pull request by Glitchfix**
add CSI mount path to portworx container

Signed-off-by: Shivanjan Chakravorty <schakravorty@purestorage.com>

**What this PR does / why we need it**:

With the new installation method with operator, CSI socket path is not mounted in portworx container.
The fix for this was to include the CSI volume mount path in while creating portworx container

### CODE REVIEW

File: drivers/storage/portworx/deployment.go [hunk 1]
```diff
@@ -1462,6 +1463,7 @@ func (t *template) getCSIVolumeInfoList() []volumeInfo {
 		volumeInfo{
 			name:         "csi-driver-path",
 			hostPath:     t.csiConfig.DriverBasePath(),
+			mountPath:    t.csiConfig.DriverBasePath(),
```
Comment (haibinxie) [inline drivers/storage/portworx/deployment.go:1466]: as we discussed in the meeting, the kubeletPath should be retrieved from env var.

**Comment by zoxpx** [comment]
**ATTN**: This commit breaks Portworx installs for all non-OpenShift Kubernetes deployments !!

The issue is that we are adding a new `/var/lib/kubelet/plugins/pxd.portworx.com` mount, which will get rejected since the parent kubelet-directory `/var/lib/kubelet` is already (automatically) mounted.
* we added code into `px-runc` binary to specifically check for such mount-overlaps, because they can lead into very bad consequences  (e.g. the host-system will hang, after `portworx.service` is restarted 5 to 10 times).

**Comment by haibinxie** [comment]
> **ATTN**: This commit breaks Portworx installs for all non-OpenShift Kubernetes deployments !!
> 
> The issue is that we are adding a new `/var/lib/kubelet/plugins/pxd.portworx.com` mount, which will get rejected since the parent kubelet-directory `/var/lib/kubelet` is already (automatically) mounted.
> 
> * we added code into `px-runc` binary to specifically check for such mount-overlaps, because they can lead into very bad consequences  (e.g. the host-system will hang, after `portworx.service` is restarted 5 to 10 times).

what's your suggestion to fix, shall oci-mon take a change too?

**Comment by zoxpx** [comment]
> what's your suggestion to fix, shall oci-mon take a change too?

The original issue was with a PX+CSI deployments on IBM's K8s clusters, which use a non-standard kubelet directory (~`/var/data/kubelet/`~ instead of `/var/lib/kubelet`).
As a better fix, I would suggest to detect the destination-cluster type as IBM-cluster, and change the CSI socket's location  (i.e. `env CSI_ENDPOINT=/var/data/kubelet/csi-plugins/pxd.portworx.com/csi.sock`)

**Comment by vinayakshnd** [comment]
@Glitchfix pls close this PR in favor of correct fix done as part of https://github.com/libopenstorage/operator/pull/884

**Pull request closed without merging**